openmdao/devtools/xdsm_viewer/xdsm_writer.py

```diff
@@ -789,21 +867,25 @@ def _prune_connections(conns, model_path=None, sep='.'):
         return conns, external_inputs, external_outputs
     else:
         for conn in conns:
-            src = conn['src']
+            src = src0 = conn['src']
+            if src.startswith(model_path):
+                src = src[len(model_path):]
             src_path = _format_name(src.rsplit(sep, 1)[0])
-            tgt = conn['tgt']
+            tgt = tgt0 = conn['tgt']
+            if tgt.startswith(model_path):
+                tgt = tgt[len(model_path):]
             tgt_path = _format_name(tgt.rsplit(sep, 1)[0])
 
-            if src.startswith(model_path) and tgt.startswith(model_path):
+            if src0.startswith(model_path) and tgt0.startswith(model_path):
                 # Internal connections
                 internal_conns.append({'src': src_path, 'tgt': tgt_path})
-            elif not src.startswith(model_path) and tgt.startswith(model_path):
+            elif not src0.startswith(model_path) and tgt0.startswith(model_path):
                 # Externally connected input
                 external_inputs.append({'src': src_path, 'tgt': tgt_path})
-            elif src.startswith(model_path) and not tgt.startswith(model_path):
+            elif src0.startswith(model_path) and not tgt0.startswith(model_path):
                 # Externally connected output
                 external_outputs.append({'src': src_path, 'tgt': tgt_path})
-
+        print('CONNNNNS', internal_conns, external_inputs, external_outputs)
```

Comment:
looks like leftover debugging?

Reply:
ah  it looks like your new PR replaces this one (and this is cleaned up there)

Reply:
Oh, thanks. I will change it in the new one.